### ISSUE TITLE
Can't install addons-docs: dependency conflict with angular

### ISSUE THREAD

**Issue by arsinclair** [storybookjs/storybook#16290]
**Describe the bug**
Unable to add _@storybook/addon-docs_ to my project using npm. It reports a dependency conflict with angular package, however the components I'm using the storybook for are all written in React and I shouldn't need the angular package.

After I run `npm install -D @storybook/addon-docs`, the following error is returned:

> npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! dev @storybook/addon-docs@"*" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @storybook/angular@6.3.10
npm ERR! node_modules/@storybook/angular
npm ERR!   peerOptional @storybook/angular@"6.3.10" from @storybook/addon-docs@6.3.10
npm ERR!   node_modules/@storybook/addon-docs
npm ERR!     dev @storybook/addon-docs@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/jrm/.npm/eresolve-report.txt for a full report.
>
> npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jrm/.npm/_logs/2021-10-09T18_30_14_847Z-debug.log

If I run `npm install -D @storybook/addon-docs --force`, the package is installed, but as I understand, forcing it is not the most orthodox thing to do.

**System**
> Environment Info:
> 
>  System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (6) x64 Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
  Binaries:
    Node: 14.17.3 - ~/.nvm/versions/node/v14.17.3/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.17.3/bin/yarn
    npm: 7.19.1 - ~/.nvm/versions/node/v14.17.3/bin/npm
  npmPackages:
    @storybook/addon-backgrounds: ^6.3.10 => 6.3.10 
    @storybook/addon-controls: ^6.3.10 => 6.3.10 
    @storybook/addons: ^6.3.10 => 6.3.10 
    @storybook/react: ^6.3.10 => 6.3.10 

**Additional context**
Relevant sections from my _package.json_:

```json
  "dependencies": {
    "gatsby-image": "^3.11.0",
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "styled-components": "^5.3.1"
  },
  "devDependencies": {
    "@babel/cli": "^7.15.7",
    "@babel/core": "^7.15.8",
    "@babel/preset-react": "^7.14.5",
    "@storybook/addon-backgrounds": "^6.3.10",
    "@storybook/addon-controls": "^6.3.10",
    "@storybook/addons": "^6.3.10",
    "@storybook/react": "^6.3.10",
    "babel-loader": "^8.2.2"
  },
```

**Comment by arsinclair** [comment]
@shilman this is good to know. I did search around and I found a similar issue but with different dependency conflict, so I wasn't sure if it's related. The `--force` workaround seems suitable for now.

Is NPM going to recognise optional dependencies at some point?

**Comment by ndelangen** [comment]
This is fixed in storybook 7.0 beta.
